nixos/modules/programs/thunderbird.nix

```diff
@@ -0,0 +1,97 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+let
+  cfg = config.programs.thunderbird;
+  policyFormat = pkgs.formats.json { };
+  thunderbirdVersion = "115";
+  policyDoc = "https://github.com/thunderbird/policy-templates/tree/master/templates/${thunderbirdVersion}";
+in
+{
+  options.programs.thunderbird = {
+    enable = lib.mkEnableOption "Thunderbird mail client";
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.thunderbird;
+      description = "Thunderbird package to use.";
+      defaultText = lib.literalExpression "pkgs.thunderbird";
+    };
+
+    policies = lib.mkOption {
+      type = policyFormat.type;
+      default = { };
+      description = ''
+        Group policies to install.
+
+        See [Thunderbird's documentation](${policyDoc})
+        for a list of available options.
+
+        This can be used to install extensions declaratively! Check out the
+        documentation of the `ExtensionSettings` policy for details.
+
+      '';
+    };
+
+    preferences = lib.mkOption {
+      type =
+        with lib.types;
+        attrsOf (oneOf [
+          bool
+          int
+          str
+        ]);
+      default = { };
+      description = ''
+        Preferences to set from `about:config`.
+
+        Some of these might be able to be configured more ergonomically
+        using policies.
+      '';
+    };
+
+    preferencesStatus = lib.mkOption {
+      type = lib.types.enum [
+        "default"
+        "locked"
+        "user"
+        "clear"
+      ];
+      default = "locked";
+      description = ''
+        The status of `thunderbird.preferences`.
+
+        `status` can assume the following values:
+        - `"default"`: Preferences appear as default.
+        - `"locked"`: Preferences appear as default and can't be changed.
+        - `"user"`: Preferences appear as changed.
+        - `"clear"`: Value has no effect. Resets to factory defaults on each startup.
+      '';
+    };
+  };
+
+  config = {
+    environment.systemPackages = [ cfg.package ];
+
+    environment.etc =
+      let
+        policiesJSON = policyFormat.generate "thunderbird-policies.json" { inherit (cfg) policies; };
+      in
+      lib.mkIf (cfg.policies != { }) { "thunderbird/policies/policies.json".source = "${policiesJSON}"; };
+
+    programs.thunderbird.policies = {
+      DisableAppUpdate = true;
```

Comment:
I think this should be shown in the options documentation, we can let `programs.thunderbird.policies` be a freeform submodule and explicitly add `DisableAppUpdate` with `default = false;`.

Reply:
Couldn't this cause issues if we let Thunderbird update itself?

Reply:
But wouldn't allowing Thunderbird to update itself cause problems?

Reply:
https://github.com/NixOS/nixpkgs/blob/cab94ab46e676b5c6aaf98ad602d9b8ed14191dd/nixos/modules/programs/firefox.nix#L284C1-L287C7

Firefox has the same approach

Reply:
Ah sorry, I meant `true`. My bad.

Reply:
Yeah, it's only that or `Status` with is overridable with `preferencesStatus` or manually. Thanks for the approval